commons/servers/DAOClient.ts

```diff
@@ -21,31 +15,54 @@ export class DAOClient {
   }
 
   async getAllServers(): Promise<Set<ServerMetadata>> {
-    const result: Set<ServerMetadata> = new Set();
+    // Check count on the list
+    const count = await this.contract.getCount()
 
-    const count = parseInt(await this.contract.methods.catalystCount().call());
+    // Create a new list
+    const newServers: Map<CatalystId, ServerMetadata> = new Map()
 
     for (let i = 0; i < count; i++) {
-      try {
-        const katalystId = await this.contract.methods.catalystIds(i).call();
-        let { id, owner, domain } = await this.contract.methods.catalystById(katalystId).call();
+      // Find id in index
+      const id = await this.contract.getCatalystIdByIndex(i)
 
-        domain = domain.trim()
+      // Check if id is known
+      let metadata = this.servers.get(id)
 
-        if (domain.startsWith("http://")) {
-          console.warn(`Catalyst node domain using http protocol, skipping ${domain}`);
-          continue;
-        }
+      // If it isn't known, then calculate it
+      if (!metadata) {
+        const data = await this.contract.getServerData(id)
+        metadata = this.toMetadata(data)
+      }
 
-        if (!domain.startsWith("https://")) {
-          domain = "https://" + domain;
-        }
+      // If metadata is defined, then store it
+      if (metadata) {
```

Comment:
Could this be undefined/null? Let's remove it if not.

Reply:
It can, that's why we are checking